base/tks/src/main/java/org/dogtagpki/server/tks/servlet/SecureChannelProtocol.java

```diff
@@ -1003,9 +1010,19 @@ public SymmetricKey unwrapSymKeyOnToken(CryptoToken token, SymmetricKey unwrappi
         }
 
         try {
-            encryptor = token.getCipherContext(EncryptionAlgorithm.DES3_ECB);
+            EncryptionAlgorithm  encAlg = EncryptionAlgorithm.DES3_ECB;
+            KeyWrapAlgorithm wrapAlg = KeyWrapAlgorithm.DES3_ECB;
+
+            IVParameterSpec iv = null;
+            if(encUnwrapKey.getType() == SymmetricKey.Type.AES) {
+                encAlg = EncryptionAlgorithm.AES_128_CBC_PAD;
```

Comment:
this is hardcoded.  But I suppose it's okay for first cut.

Reply:
Noted. We can change this whenever we want.